code/drasil-lang/lib/Language/Drasil/Expr/Extract.hs

```diff
@@ -76,4 +76,4 @@ eNamesRI' (UpFrom il)     = eNames' (snd il)
 
 -- | Get dependencies from an equation.  
 eDep :: Expr -> [UID]
-eDep = nub . eNames
+eDep = toList . fromList . eNames
```

Comment:
@NoahCardoso If you're struggling with figuring out which things should be Sets and which should be lists, I think there are two ways you can go about this:
1. Research first.
2. Make a hypothesis and test it.

In other words,
1. Traverse over all things that use `eDep`, and then all things that use those things, and figure out if order is somehow important and why it is (or why it shouldn't be...). Thankfully, there isn't too much usage of `eDep`, this might be easier and better, logically.
2. Semi-confidently assume that this function should return a Set rather than a list. Our hypothesis of sorts -- the comment just says "Get dependencies from an equation," but it doesn't mention any sort of ordering of the dependencies, it just lists what a specific expression depends on (i.e., variables and functions). You can try this, and see what breaks.

Similar to `eDep`, I would imagine this extends to `sdep`, `shortdep`, `eNames`, but you can figure it out as you go along.

<details>

<summary>I took a quick peek at `eDep` usage. It doesn't like there's too much to traverse</summary>

```console
λ ~/Programming/Drasil/ glassbrIntro rg "[^a-z]eDep" -ths
code/drasil-code/lib/Language/Drasil/Chunk/CodeBase.hs
20:codevars e m = map (varResolve m) $ eDep e
24:codevars' e m = map (varResolve m) $ nub $ eDep' e

code/drasil-lang/lib/Language/Drasil/Expr/Development.hs
9:  , eDep, eNames, eNames', eNamesRI
15:import Language.Drasil.Expr.Extract (eDep, eNames, eNames', eNamesRI)

code/drasil-lang/lib/Language/Drasil/CodeExpr/Extract.hs
2:    eDep, eDep',

code/drasil-lang/lib/Language/Drasil/CodeExpr/Development.hs
11:    eDep, eDep',
25:import Language.Drasil.CodeExpr.Extract (eDep, eDep', eNamesRI, eNamesRI')
```

</details>

Reply:
eDep is only used in codevars. I have eDep as a Set currently. This is roughly the order of the other functions that relate to eDep. eDep -> codevars -> fstvars -> fstdecl -> genFunc -> genModDef, genModFuncs ... -> ... generates things. Once things start to get generated in gool I assume that order matters. Codervars, fstvars and fstdecl might be able to be Sets however codevars is a list of CodeChunks which are not `ord`

Reply:
Also `meDep` is used for printing. Should it be? I thought it was only for getting dependencies and its order shouldn't matter.
https://github.com/JacquesCarette/Drasil/blob/864e1385e94f900d8035f605a0b170b9c439dc61/code/drasil-lang/lib/Language/Drasil/ModelExpr/Extract.hs#L87-L89